src/passes/LogExecution.cpp

```diff
@@ -39,23 +40,47 @@ namespace wasm {
 Name LOGGER("log_execution");
 
 struct LogExecution : public WalkerPass<PostWalker<LogExecution>> {
-  void visitLoop(Loop* curr) { curr->body = makeLogCall(curr->body); }
+  std::map<Function*, Index> functionOrdinals;
 
-  void visitReturn(Return* curr) { replaceCurrent(makeLogCall(curr)); }
+  Index nextFreeIndex = 0;
+
+  // Tries to convert a string to a function index. Returns (Index)-1 on
+  // failure.
+  Index stringToIndex(const char* s) {
+    for (const char* q = s; *q; ++q)
+      if (!isdigit(*q))
+        return (Index)-1;
```

Comment:
Binaryen style is to always use braces.
```suggestion
    for (const char* q = s; *q; ++q) {
      if (!isdigit(*q)) {
        return (Index)-1;
      }
    }
```